### DISCUSSION TITLE
Language translation with using "Translation Strings As Keys"

### DISCUSSION THREAD

**Discussion by dilarakanalici** [Okipa/laravel-table#124] [Q&A]
Everything is okay when I make lang/messages file and reference to them but json file seems like can not read by laravel-table. Is there anyone had same issue or am I making a mistake in somewhere? Thanks for all helps.

**Comment by Okipa** [comment]
Hi @dilarakanalici, I never faced this issue on my end with this package, sorry!

Each word and sentence used by `laravel-table` is referenced [here](https://github.com/Okipa/laravel-table#translations) and is translated this way: `__('Loading in progress...')`.

I think that your issue has nothing to do with this package but is related to the way you manage your translations in your project.

Did you also set any translations using dot notation like this `__('user.action.unauthorized')` for example?

I had the case on a project once when I mixed those two translation ways that a conflict occurred between a `_('User')` translation and a `__('user.<whatever>')` dot translation. Could it be related to something like that?

**Reply by dilarakanalici** [reply]
Thanks for quick answer but I don't think so. I guess my project completely ignored lang/{lang_code}.json files. I am using laravel 10 If you want to know.

**Reply by Okipa** [reply]
@dilarakanalici, in case you did a migration, the directory containing the lang files has changed with L10.

It is not located in the `resources` directory anymore but must be located directly at the root of the repository.

Would this help?

**Reply by dilarakanalici** [reply]
Yes, from the very beginning, it is in the root directory.
